source/extensions/quic_listeners/quiche/platform/quic_mem_slice_impl.cc

```diff
@@ -0,0 +1,51 @@
+// NOLINT(namespace-envoy)
+//
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "extensions/quic_listeners/quiche/platform/quic_mem_slice_impl.h"
+
+#include "common/common/assert.h"
+
+namespace quic {
+
+Envoy::Buffer::BufferFragmentImpl*
+QuicMemSliceImpl::allocateBufferAndFragment(QuicBufferAllocator* allocator, size_t length) {
+  auto bundle = reinterpret_cast<BufferFragmentBundle*>(
+      allocator->New(sizeof(BufferFragmentBundle) + length));
+  Envoy::Buffer::BufferFragmentImpl& fragment = bundle->fragment_with_padding_.fragment_;
+  return new (&fragment) Envoy::Buffer::BufferFragmentImpl(
+      bundle->buffer_, length,
+      [allocator](const void*, size_t, const Envoy::Buffer::BufferFragmentImpl* frag) {
+        // Delete will deallocate the bundle of fragment and buffer.
```

Comment:
What calls Envoy::Buffer::BufferFragmentImpl's destructor?

Reply:
BufferFragmentImpl and the following buffer_ is allocated together via allocator->New() as an array of their total size and they are deallocated via allocator->Delete() as an array too. see QuicBufferAllocator interface; https://quiche.googlesource.com/quiche/+/ff86db3f4a3d59250eeafbb9787fb52b70e6ac31/quic/core/quic_buffer_allocator.h?autodive=0%2F%2F